### PR TITLE
Changes after initial code review

### PR DESCRIPTION
**Pull request by regulskimichal**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,33 +1,221 @@
...
-.classpath
-.factorypath
-.project
-.settings
-.springBeans
-.sts4-cache
-
-### IntelliJ IDEA ###
-.idea
+### JetBrains template
```
Comment (regulskimichal) [inline .gitignore:1]: A bit more complex .gitignore covering more cases

File: .run/Compose Deployment.run.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<component name="ProjectRunConfigurationManager">
```
Comment (regulskimichal) [inline .run/Compose Deployment.run.xml:1]: These files should allow sharing configurations for running different scripts/applications from that project in IntelliJ IDEA

File: build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+plugins {
+    id("org.springframework.boot") version "2.5.2"
```
Comment (regulskimichal) [inline build.gradle.kts:2]: Defines versions of dependencies used in the Spring Boot project. https://docs.spring.io/spring-boot/docs/current/reference/html/dependency-versions.html

File: build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    id("io.spring.dependency-management") version "1.0.11.RELEASE"
```
Comment (regulskimichal) [inline build.gradle.kts:3]: Cares about using proper versions in the project. Allows skipping defining version in the dependencies block.

File: docker-compose.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+version: "3"
+
+services:
+  postgres:
```
Comment (regulskimichal) [inline docker-compose.yaml:4]: Defines containerized database

File: docker-compose.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    image: postgres:13-alpine
```
Comment (regulskimichal) [inline docker-compose.yaml:5]: Alpine Linux is a very minimal distro of Linux. Works nicely in containers.

File: src/main/java/com/wishlist/Context.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.springframework.http.HttpMethod.*;
+
+@Configuration
+public class Context {
+
+    @Bean
+    public CorsFilter corsFilter() {
```
Comment (regulskimichal) [inline src/main/java/com/wishlist/Context.java:19]: Needs more work

File: src/main/java/com/wishlist/controllers/ProductController.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+import com.wishlist.model.Product;
+import com.wishlist.services.ProductService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/product")
```
Comment (regulskimichal) [inline src/main/java/com/wishlist/controllers/ProductController.java:13]: In REST controllers we want to expose different resources in a standardized way. Read more about it:
https://www.restapitutorial.com/lessons/httpmethods.html
https://www.restapitutorial.com/lessons/restfulresourcenaming.html
https://docs.microsoft.com/en-us/azure/architecture/best-practices/api-design
https://stackoverflow.blog/2020/03/02/best-practices-for-rest-api-design/

File: src/main/java/com/wishlist/exceptions/GlobalExceptionHandler.java [hunk 9]
```diff
@@ -0,0 +1,17 @@
+package com.wishlist.exceptions;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
+@Slf4j
+public class GlobalExceptionHandler {
```
Comment (regulskimichal) [inline src/main/java/com/wishlist/exceptions/GlobalExceptionHandler.java:10]: It is a good idea to handle all business exceptions in one place.

Comment (regulskimichal) [reply src/main/java/com/wishlist/exceptions/GlobalExceptionHandler.java:10]: Or at least in few places, but one exception should always be handled in the same way.

File: src/main/java/com/wishlist/exceptions/ProductNotFoundException.java [hunk 10]
```diff
@@ -0,0 +1,9 @@
+package com.wishlist.exceptions;
+
+public class ProductNotFoundException extends RuntimeException {
+
```
Comment (regulskimichal) [inline src/main/java/com/wishlist/exceptions/ProductNotFoundException.java:4]: The message does not need to be stored as a field. We just want to pass it to the super class.

File: src/main/java/com/wishlist/model/Product.java [hunk 11]
```diff
@@ -0,0 +1,25 @@
+package com.wishlist.model;
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.lang.NonNull;
+
+import javax.persistence.*;
+import java.io.Serializable;
+
+@Data
```
Comment (regulskimichal) [inline src/main/java/com/wishlist/model/Product.java:10]: Data covers Getter Setter RequiredArgsConstructor ToString EqualsAndHashCode

File: src/main/java/com/wishlist/model/Product.java [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+@NoArgsConstructor
```
Comment (regulskimichal) [inline src/main/java/com/wishlist/model/Product.java:11]: Required by JPA

File: src/main/java/com/wishlist/model/Product.java [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+@Entity
+@Table(name = "product")
```
Comment (regulskimichal) [inline src/main/java/com/wishlist/model/Product.java:13]: Some SQL dialects require in SQLs to use quote marks when table name has capital letters

File: src/main/java/com/wishlist/repositories/ProductRepository.java [hunk 14]
```diff
@@ -0,0 +1,8 @@
+package com.wishlist.repositories;
+
+import com.wishlist.model.Product;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface ProductRepository extends JpaRepository<Product, Long> {
```
Comment (regulskimichal) [inline src/main/java/com/wishlist/repositories/ProductRepository.java:6]: JpaRepository has bigger functionality

**Review by PawelRozniecki — APPROVED** [review]

File: src/main/java/com/wishlist/services/ProductService.java [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+import com.wishlist.exceptions.ProductNotFoundException;
+import com.wishlist.model.Product;
+import com.wishlist.repositories.ProductRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Service
+@Transactional(readOnly = true)
```
Comment (regulskimichal) [inline src/main/java/com/wishlist/services/ProductService.java:12]: Transactional(readOnly = true) on top of the class makes all methods transactional (of course on the database level), but in the transactions no write is possible. Modifying methods should be annotated explicitly with Transational.

File: src/main/resources/application.yaml [hunk 16]
```diff
@@ -0,0 +1,16 @@
+spring:
```
Comment (regulskimichal) [inline src/main/resources/application.yaml:1]: yaml format seems to me more readable and cleaner.

File: src/test/java/com/wishlist/StoreApplicationTests.java [hunk 17]
```diff
@@ -1,4 +1,4 @@
-package com.pwr.AngularSpring.Store;
+package com.wishlist;
```
Comment (regulskimichal) [inline src/test/java/com/wishlist/StoreApplicationTests.java:1]: package names shouldn't contain capital letters

**Pull request merged**